### ISSUE TITLE
Make the way to get intel-formatted assembly from rustc more discoverable

### ISSUE THREAD

**Issue by scottmcm** [rust-lang/rust#48011]
As far as I can tell, nowhere in `rustc --help` does it say how to control the ASM syntax.  I only found it by finding this random gist: https://gist.github.com/bluss/5a088d3f420d12406689439e2940d731

Brainstorming ways to make it more discoverable:
- add an `--emit=asm-intel` option
- document `-C "llvm-args=-x86-asm-syntax=intel"` in the help somewhere
- ???

**Comment by nagisa** [comment]
> add an --emit=asm-intel option

can't work because the option is x86-specific.

**Comment by rkruppe** [comment]
We could make that parameter value only valid for x86 targets right? Not very elegant, but since this is a target-specific concern, there's no way to avoid having something target specific in the CLI (either the availability of an option, or the validity of some value).

I'm not a fan of `--emit asm-intel` though, because different emit types should be able to exist side by side, but both `--emit asm` and `--emit asm-intel` would naturally want to output `foo.s`. Perhaps a codegen or `-Z` option that changes the flavor of `--emit asm`? We already have `-Zasm-comments`, for comparison.

**Comment by trondsg** [comment]
> 



> but both `--emit asm` and `--emit asm-intel` would naturally want to output `foo.s`

No, because the extension normally used for intel assembly is .asm, not .s.
